src/backend/fastapi_app/rag_simple.py

```diff
@@ -1,16 +1,95 @@
 import pathlib
+from abc import ABC, abstractmethod
 from collections.abc import AsyncGenerator
 from typing import Any
 
-from openai import AsyncAzureOpenAI, AsyncOpenAI
-from openai.types.chat import ChatCompletion, ChatCompletionMessageParam
+from openai import AsyncAzureOpenAI, AsyncOpenAI, AsyncStream
+from openai.types.chat import ChatCompletion, ChatCompletionChunk, ChatCompletionMessageParam
 from openai_messages_token_helper import build_messages, get_token_limit
+from pydantic import BaseModel
 
-from .api_models import Message, RAGContext, RetrievalResponse, ThoughtStep
-from .postgres_searcher import PostgresSearcher
+from fastapi_app.api_models import (
+    AIChatRoles,
+    Message,
+    RAGContext,
+    RetrievalResponse,
+    RetrievalResponseDelta,
+    ThoughtStep,
+)
+from fastapi_app.postgres_models import Item
+from fastapi_app.postgres_searcher import PostgresSearcher
 
 
-class SimpleRAGChat:
+class ChatParams(BaseModel):
+    top: int = 3
+    temperature: float = 0.3
+    response_token_limit: int = 1024
+    enable_text_search: bool
+    enable_vector_search: bool
+    original_user_query: str
+    past_messages: list[ChatCompletionMessageParam]
+    prompt_template: str
+
+
+class RAGChatBase(ABC):
+    current_dir = pathlib.Path(__file__).parent
+    query_prompt_template = open(current_dir / "prompts/query.txt").read()
+    answer_prompt_template = open(current_dir / "prompts/answer.txt").read()
+
+    def get_params(self, messages: list[ChatCompletionMessageParam], overrides: dict[str, Any]) -> ChatParams:
+        top: int = overrides.get("top", 3)
```

Comment:
I inherited this from azure-search-openai-demo, but I wonder if there's a nicer way to get these parameters, if we have more typing on the overrides, with default values in the API model?

Reply:
That would also mean ChatParams was moved to api_models.py